src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -193,110 +246,93 @@ export function useSwapInputsController({
         }
       }
 
-      isFetching.value = false;
       quote.value = data;
-      runOnJS(updateQuoteStore)(data);
 
       if (!data || (data as QuoteError)?.error) {
         return;
       }
 
       if (inputAmount) {
-        const price = internalSelectedInputAsset.value?.nativePrice || internalSelectedInputAsset.value?.price?.value || 0;
+        const price = inputPrice || inputNativePrice.value;
         inputValues.modify(prev => {
+          updateNativePriceForAsset({ price, type: 'inputAsset' });
```

Comment:
additionally: why is this `updateNativePriceForAsset` inside the modify function itself of the `inputValues`; everything here is being run synchronously on the UI thread so not sure why it needs to be inside the `modify` (same for outputAmount below)